src/main/java/sparksoniq/spark/iterator/flowr/ReturnClauseSparkIterator.java

```diff
@@ -80,9 +80,9 @@ protected Item nextLocal() {
 
     @Override
     protected void openLocal(DynamicContext context) {
-        _child.open(context);
-        _isExpressionOpen = false;
         _tupleContext = new DynamicContext(_currentDynamicContext);     // assign current context as parent
+        _child.open(_tupleContext);
```

Comment:
here too, _child.open(context) was fine. _tupleContext is meant to be the context updated in place for use on _expression.